### ISSUE TITLE
Wrong extraction of text layer over a scanned image

### ISSUE THREAD

**Issue by witchi** [smalot/pdfparser#60]
I try to use the search_lucene app for Owncloud 8 (https://github.com/owncloud/search_lucene), which uses your library. If I upload PDFs into the cloud, the PDFParser should return the text of the PDF. But all of the text is one lump without word separation. The PDF comes from https://github.com/fritz-hh/OCRmyPDF, which translates a scanned document into a layered PDF (1.4). I have tested it with your demo page and it seems, that the missing whitespace prevents the usage of the fulltext indexer of Owncloud. 

My PDF stores boxes around each word and places the boxes in a transparent layer over the scanned image. So it seems, that your library extracts the correct text content, but merges all the text parts together without any separator. Is it possible to enhance the lib a little bit to get "readable" text from the layered PDF?

**Comment by witchi** [comment]
You can download the PDF from 

https://drive.google.com/open?id=0B8F9pu5KRD3Yfl9nZFdQTkNyemJSMmFzNDJkQmxOMDQtZmxnWl9OZ1R2TnVmbGJpcDZhNjA&authuser=0

It is the printed README of your project, scanned with 300dpi and OCR'ed and translated into PDF. In a PDF viewer you can "Select all" to see the transparent text layer.

**Comment by witchi** [comment]
The output of your PDFParser's Demo page is

```
pdfparser/READMEmdatmaster'smalot/pdfparserhttpszl/github.com/smaloflpdfparser/blob/master/R...
0Thisrepositoryaus:ExploreGlstBlogHelpwltchl+vCaQI3’
SmaIOt/pdfparserQwatchv3s‘tstar211VForka3
Farnemastervpdfparser/READMEmd
o
EsmalotoneMay2014UpdateREADMEmd
2contributorsi‘G)
I‘!
42IInes(26sIoc)1.301kbRawBlameHlstoryfmfit?
PdfParser‘s
PdfParser,astandalonePHPiibrary,providesvarioustoolstoextractdatafromaPDFfiie.
buridpasminq
Website:httpJ/wwwpdfparsenorg
TesttheAPIonourdemopage.
ThisprojectissupportedbyActuaiys.
Features
Featuresinciuded:
0Load/parseobjectsandheaders
0Extractmetadata(author,description,...)0Extracttextfromorderedpages0Supportofcompressedpdf0SupportofMACOSRomancharsetenoodingoHandlingofhexaandoctaiencodingintoxtsections
0PSR-Ooompliant(autoioader)CPSR-icompliant(Codestyiing)
Currentiy,secureddocumentsarenotsupported.
ThisLibraryisstillunderactivedeveiopment.Asaresult‚usersmustexpectBCbreakswhenusingthemasterVersion.
Documentation
ReadthedocumentationonWebsite.
License
ThisiibraryisundertheGPLv3license.
02.015GitHub‚Inc.TermsPrivacySecurnyContact0StatusAPITrainingShop850gAbout
1von125.03.201512:55 
```

Some words, like "Website" can be recognized, other words like "Library" cannot be found.

**Comment by witchi** [comment]
For the moment I have added an `else` branch after line 327 in Object.php:

```
} else {
    $text .= " ";
}
```
which adds a space character after almost every "Tm" command to the output. It works for my PDF, but there could be side-effects on other files. I don't have any knowledge about the PDF standard.

**Comment by Miliamber** [comment]
+1 here workaround from @witchi works great

**Comment by Connum** [comment]
I don't think I'll work on this, as I think there are more pressing issues, but I wanted to post a quick update for anyone wanting to take this on: 

The issue doesn't present itself as gravely with the current version as it did 5 years ago. Almost all of the words are now separated by (at least) one space, most actually even have heavy spacing in between them. Only some words keep sticking together, and the fix provided by @witchi still works for those.

But I'm not too deep into the entirety of the library's code or the PDF spec to really understand the impact this fix could have, or why it works at all.

**Comment by k00ni** [comment]
We had a lot of fixes lately, please try again (`v2.10.0+`) and get back to us, if the problem persists.
